aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -44,6 +44,14 @@ IntList SparseTensorImpl::sizes() const {
 IntList SparseTensorImpl::strides() const {
   AT_ERROR("sparse tensors do not have strides");
 }
+int64_t SparseTensorImpl::size(int64_t d) const {
+  d = at::maybe_wrap_dim(d, dim(), false);
+  return size_[d];
+}
+int64_t SparseTensorImpl::stride(int64_t d) const {
+  AT_ERROR("sparse tensors do not have strides");
```

Comment:
Yeah, we can devirtualize stride if we have some boolean flag we can check if the tensor understands strides or not. Not this PR anyway.